apptools/persistence/state_pickler.py

```diff
@@ -341,8 +360,19 @@ def _do(self, obj):
             return self._do_dict(obj)
         elif hasattr(obj, '__dict__'):
             return self._do_instance(obj)
+        else:
+            warnings.warn("Cannot pickle unrecognized type {}. Returning None"
+                          " for backward compatibility.".format(obj_type))
+            return None
 
     def _get_id(self, value):
+        # We consider a special case for numpy scalar values, because
+        # they hash as native types, but they are special because we
+        # want to recover their true type, and the only way of doing so
+        # is to consider them as objects.
+        if isinstance(value, numpy.generic):
+            return id(value)
```

Comment:
I have updated the pickling do method to keep the object alive in the cash (for the duration of the pickling process). This should be enough to avoid the `id` reuse case.